hudi-client/hudi-client-common/src/test/java/org/apache/hudi/io/TestSecondaryIndexStreamingTracker.java

```diff
@@ -0,0 +1,709 @@
+   * Then: A single secondary index entry should be added with isDeleted=false
+   */
+  @Test
+  public void testTrackSecondaryIndexStats_InsertNewRecord() {
+    // Create a HoodieKey
+    String recordKey = "test-record-key-001";
+    String partitionPath = "2023/01/01";
+    HoodieKey hoodieKey = new HoodieKey(recordKey, partitionPath);
+    
+    // Create a new record with a secondary index field value
+    GenericRecord avroRecord = new GenericData.Record(schema);
+    
+    // Set Hoodie metadata fields
+    avroRecord.put("_hoodie_record_key", recordKey);
+    avroRecord.put("_hoodie_partition_path", partitionPath);
+    avroRecord.put("_hoodie_commit_time", "20231201120000");
+    avroRecord.put("_hoodie_commit_seqno", "001");
+    avroRecord.put("_hoodie_file_name", "test-file.parquet");
+    
+    // Set data fields
+    avroRecord.put("id", recordKey);
+    avroRecord.put("name", "Test User");
+    avroRecord.put("fare", 100.5); // This will be our secondary index field
+    avroRecord.put("timestamp", System.currentTimeMillis());
+    
+    HoodieAvroPayload payload = new HoodieAvroPayload(Option.of(avroRecord));
+    HoodieRecord newRecord = new HoodieAvroRecord<>(hoodieKey, payload);
+    
+    // Create secondary index definition for the "fare" field
+    HoodieIndexDefinition indexDef = HoodieIndexDefinition.newBuilder()
+        .withIndexName(MetadataPartitionType.SECONDARY_INDEX.getPartitionPath() + "_fare_index")
+        .withIndexType("SECONDARY_INDEX")
+        .withSourceFields(Collections.singletonList("fare"))
+        .build();
+    
+    // Test case: Insert a new record (no old record exists)
+    SecondaryIndexStreamingTracker.trackSecondaryIndexStats(
+        hoodieKey,
+        Option.of(newRecord),
+        null, // No old record
+        false, // Not a delete
+        writeStatus,
+        schema,
+        () -> schema,
+        Arrays.asList(indexDef),
+        Option.empty(), // No key generator needed for this test
+        config
+    );
+    
+    // Verify that a secondary index entry was added (not deleted)
+    assertFalse(writeStatus.getIndexStats().getSecondaryIndexStats().isEmpty());
+    
+    // Check that the entry is an insert (isDeleted = false)
+    writeStatus.getIndexStats().getSecondaryIndexStats().forEach((indexName, statsList) -> {
+      assertEquals(1, statsList.size());
+      statsList.forEach(stats -> {
+        assertEquals(recordKey, stats.getRecordKey());
+        assertEquals("100.5", stats.getSecondaryKeyValue()); // The fare value
+        assertFalse(stats.isDeleted()); // isDeleted should be false for insert
+      });
+    });
+  }
+
+  /**
+   * Given: An old record and a new record with the same secondary key value
+   * When: trackSecondaryIndexStats compares the records
+   * Then: No secondary index updates should be recorded (shouldUpdate=false optimization)
+   */
+  @Test
+  public void testTrackSecondaryIndexStats_UpdateWithSameSecondaryKey() {
```

Comment:
can we parametrize these tests. we just need diff values for old record and new record right. otherwise entire test str seems duplicated